LEGO1/lego/legoomni/src/race/legoracecar.cpp

```diff
@@ -18,42 +22,78 @@ LegoRaceCar::LegoRaceCar()
 	NotificationManager()->Register(this);
 }
 
-// STUB: LEGO1 0x10012c80
+// FUNCTION: LEGO1 0x10012c80
 LegoRaceCar::~LegoRaceCar()
 {
-	// TODO
+	NotificationManager()->Unregister(this);
 }
 
-// STUB: LEGO1 0x10012d90
+// FUNCTION: LEGO1 0x10012d90
 MxLong LegoRaceCar::Notify(MxParam& p_param)
 {
-	// TODO
-	return 0;
+	return LegoRaceMap::Notify(p_param);
 }
 
-// STUB: LEGO1 0x10012e60
+// FUNCTION: LEGO1 0x10012e60
 void LegoRaceCar::SetWorldSpeed(MxFloat p_worldSpeed)
 {
-	// TODO
+	if (!m_userNavFlag) {
+		if (!LegoCarRaceActor::m_unk0x0c) {
+			m_maxLinearVel = p_worldSpeed;
+		}
+		LegoAnimActor::SetWorldSpeed(p_worldSpeed);
+		return;
```

Comment:
It would be better to turn this branch into a `if..else` structure than using an early return here